app/src/main/java/com/nevilleantony/prototype/storage/StorageApi.java

```diff
@@ -93,4 +94,23 @@ public void accept(List<DownloadsModel> downloadsModels) throws Throwable {
         });
         return downloadsModels;
     }
+
+    public String insertAvailDownload(Context context, String groupid, Long part){
+        db = DownloadsDatabase.getInstance(context);
+        AvailableDownloadsModel file = new AvailableDownloadsModel(groupid, part);
+        db.getavaildao().insertAvailDownloads(file).subscribe();
+        return SUCCESS_MESSAGE;
+    }
+
+    public List<Long> retriveParts(Context context, String groupid){
```

Comment:
Can you also add a `public Map<String, List<Sting>> getAll(Context context);` which returns a hash map of each groupId as key and its parts as its values?